src/type.ts

```diff
@@ -0,0 +1,56 @@
+export interface Product {
+    id: number;
+    title: string;
+    description: string;
+    price: number;
+    images: [];
+    creationAt: string;
+    updatedAt: string;
+    category: {};
+}
+
+export interface ProductId {
+    id: number;
+    title: string;
+    price: number;
+    description: string;
+    images: [];
+    creationAt: string;
+    updatedAt: string;
+    category: {};
```

Comment:
You need to define type of your category.
export interface Product {
// ...
category: Category
}

Reply:
fixed